### PR TITLE
[WIP]: More config stuff

### PR DESCRIPTION
**Pull request by JayH5**
So far this is a bit of a "moving stuff about" PR, but at the least we get a lot of the Listener/HTTP stuff to be a little bit more configurable, `mix format`-ted, and typespec-ed.

* Pull more HTTP stuff into EnvoyUtil (EnvoyUtil will probably be split up later)
* Make some more HTTP things configurable
* Clean up config accessing

TODO:
* Think about what we do with Marathon.Adapter
* Write some docs?

### CODE REVIEW

File: lib/relay/marathon/adapter.ex [hunk 1]
```diff
@@ -21,7 +21,7 @@ defmodule Relay.Marathon.Adapter do
   """
   @spec app_clusters(App.t, keyword) :: [Cluster.t]
   def app_clusters(%App{port_indices_in_group: port_indices_in_group} = app, options \\ []) do
-    eds_config_source = EnvoyUtil.api_config_source()
+    eds_config_source = Common.api_config_source()
```
Comment (JayH5) [inline lib/relay/marathon/adapter.ex]: Not sure about calling things from `Common`.. seems a bit weird. Could do an import.

File: lib/relay/resources/common.ex [hunk 2]
```diff
@@ -0,0 +1,54 @@
+defmodule Relay.Resources.Common do
+  alias Envoy.Api.V2.Core.{Address, ApiConfigSource, ConfigSource, SocketAddress}
```
Comment (JayH5) [inline lib/relay/resources/common.ex]: Given that these things are from `Core`.. should we call our thing `Core`. OTOH, the object truncation thing is not in `Core` (or anywhere specifically).

File: lib/relay/resources/lds.ex [hunk 3]
```diff
@@ -1,22 +1,28 @@
 defmodule Relay.Resources.LDS do
+  alias Relay.{ProtobufUtil, Resources.CertInfo}
+  import Relay.Resources.Common
+
   alias Envoy.Api.V2.Listener
-  alias Relay.{EnvoyUtil, Resources.CertInfo}
+  alias Envoy.Api.V2.Listener.{Filter, FilterChain}
```
Comment (JayH5) [inline lib/relay/resources/lds.ex]: Similarly to above.. should our module be called `Listener`?

File: lib/relay/resources/common.ex (hunk 2 above)
Comment (jerith) [reply lib/relay/resources/common.ex]: We could have `R.R.Core` and `R.R.Shared`, I suppose.

**Pull request merged**